### PR TITLE
llama.cpp portable Zip for Linux quickstart

### PR DESCRIPTION
**Pull request by qiyuangong**
## Description

* llama.cpp portable Zip for Linux quickstart
* Flashmoe example

### 4. How to test?
- [ ] N/A
- [ ] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.
- [ ] Application test
- [X] Document test
- [ ] ...

### CODE REVIEW

File: docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md [hunk 1]
```diff
@@ -130,7 +137,142 @@ llama_perf_context_print:        load time =   xxxxx.xx ms
 llama_perf_context_print: prompt eval time =     xxx.xx ms /   129 tokens (    x.xx ms per token,   xxx.xx tokens per second)
 llama_perf_context_print:        eval time =   xxxxx.xx ms /  1256 runs   (   xx.xx ms per token,    xx.xx tokens per second)
 llama_perf_context_print:       total time =   xxxxx.xx ms /  1385 tokens
+```
+
+## Linux Quickstart
+
+### Prerequisites
+
```
Comment (qiuxin2012) [inline docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md:145]: follow https://github.com/intel/ipex-llm/blob/main/docs/mddocs/Quickstart/ollama_portable_zip_quickstart.md#prerequisites-1

File: docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md [hunk 2]
```diff
@@ -130,7 +137,142 @@ llama_perf_context_print:        load time =   xxxxx.xx ms
...
+### Step 1: Download and Extract
+
+Download IPEX-LLM llama.cpp portable tgz for Linux users from the [link](https://github.com/intel/ipex-llm/releases/tag/v2.2.0-nightly).
+
+Then, extract the zip file to a folder.
```
Comment (qiuxin2012) [inline docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md:150]: zip to tgz

File: docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md [hunk 3]
```diff
@@ -8,14 +8,19 @@ This guide demonstrates how to use [llama.cpp portable zip](https://github.com/i
...
   - [Prerequisites](#prerequisites)
   - [Step 1: Download and Unzip](#step-1-download-and-unzip)
   - [Step 3: Runtime Configuration](#step-2-runtime-configuration)
   - [Step 3: Run GGUF models](#step-3-run-gguf-models)
+- [Linux Quickstart](#linux-quickstart)
+  - [Prerequisites](#prerequisites-1)
+  - [Step 1: Download and Extract](#step-1-download-and-extract)
+  - [Step 2: Runtime Configuration](#step-2-runtime-configuration-1)
+  - [Step 3: Run GGUF models](#step-3-run-gguf-models-1)
+  - [(New) FlashMoE for Moe Models (e.g., DeeSeek V3/R1)](#flashmoe-for-moe-models)
```
Comment (jason-dai) [inline docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md:23]: FlashMoE for MoE Models (e.g., DeeSeek V3/R1) using llama.cpp

File: docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md [hunk 4]
```diff
@@ -130,7 +137,142 @@ llama_perf_context_print:        load time =   xxxxx.xx ms
...
+        repeat_last_n = 64, repeat_penalty = 1.000, frequency_penalty = 0.000, presence_penalty = 0.000
+        dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
+        top_k = 40, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, temp = 0.000
+        mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
+sampler chain: logits -> logit-bias -> penalties -> dry -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist
+
+generate: n_ctx = 2528, n_batch = 4096, n_predict = 2048, n_keep = 1
+```
+
+### FlashMoe for MoE Models
```
Comment (jason-dai) [inline docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md:208]: FlashMoE for DeeSeek V3/R1

File: docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md [hunk 5]
```diff
@@ -130,7 +137,142 @@ llama_perf_context_print:        load time =   xxxxx.xx ms
...
+
+sampler seed: 341519086
+sampler params:
+        repeat_last_n = 64, repeat_penalty = 1.000, frequency_penalty = 0.000, presence_penalty = 0.000
+        dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
+        top_k = 40, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, temp = 0.000
+        mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
+sampler chain: logits -> logit-bias -> penalties -> dry -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist
+
+generate: n_ctx = 2528, n_batch = 4096, n_predict = 2048, n_keep = 1
```
Comment (Oscilloscope98) [inline docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md:207]: We should at least keep some example output texts as we did for Windows? Same below.

File: docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md [hunk 6]
```diff
@@ -130,7 +137,142 @@ llama_perf_context_print:        load time =   xxxxx.xx ms
...
+- [DeepSeek-R1-Q6_K](https://huggingface.co/unsloth/DeepSeek-R1-GGUF/tree/main/DeepSeek-R1-Q6_K)
+
+#### Run DeepSeek V3/R1 with FlashMoE
+
+Requirements: 
+- 380GB CPU Memory
+- 1-8 ARC A770
+- 500GB Disk
+
+Note: larger models and other types may require more resources. For 1 ARC A770 platform, please reduce context length (e.g., 1024) to avoid OOM. Add this option `-c 1024` at the end of previous command.
```
Comment (Oscilloscope98) [inline docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md]: type -> precision?

File: docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md [hunk 7]
```diff
@@ -130,7 +137,142 @@ llama_perf_context_print:        load time =   xxxxx.xx ms
...
+
+### Step 1: Download and Extract
+
+Download IPEX-LLM llama.cpp portable tgz for Linux users from the [link](https://github.com/intel/ipex-llm/releases/tag/v2.2.0-nightly).
+
+Then, extract the zip file to a folder.
+
+### Step 2: Runtime Configuration
+
+- Open "Terminal", and enter the extracted folder through `cd /PATH/TO/EXTRACTED/FOLDER`
```
Comment (Oscilloscope98) [inline docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md:156]: Open "Terminal" -> Open a terminal?

File: docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md (hunk 5 above)
Comment (qiyuangong) [reply docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md:207]: It has been replaced with XXXX.

File: docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md [hunk 8]
```diff
@@ -203,9 +177,15 @@ sampler params:
...
+
+<think>
+XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+</think>
+
+<answer>XXXX</answer> [end of text]
 ```
 
-### FlashMoe for MoE Models
+### FlashMoe for DeeSeek V3/R1
```
Comment (jason-dai) [inline docs/mddocs/Quickstart/llamacpp_portable_zip_gpu_quickstart.md:188]: FalshMoE

**Review by Oscilloscope98 — APPROVED** [review]
Seems to have format problem in sample output for the sycl tables:
image

Others LGTM
